src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -946,6 +950,7 @@ protected BoundExpression BindInferredVariableInitializer(DiagnosticBag diagnost
             if (isVar)
             {
                 aliasOpt = null;
+                bindType = true;
```

Comment:
>bindType = true; [](start = 16, length = 16)

This doesn't look right. I don't think there is a good reason to make an exception for the ````var``` case.